### ISSUE TITLE
The config value `latex_documents` is not properly set in tests/test_build_latex.py::test_build_latex_doc

### ISSUE THREAD

**Issue by kynzie** [sphinx-doc/sphinx#6652]
**The bug**
When running the testsuite `tests/test_build_latex.py::test_build_latex_doc`, I realize that the
statement
`
app.config.latex_documents[0] = app.config.latex_documents[0][:4] + (docclass,)
` ([here](https://github.com/sphinx-doc/sphinx/blob/master/tests/test_build_latex.py#L101))
does not have the expected behavior: if `docclass` is set to `howto`, `app.config.latex_documents[0][4]` will still be set to `manual` after the statement.

This is due to:

- In `config.py` ([here](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/config.py#L254)),  `__getattr__` returns the result of `default_latex_documents` function ([here](https://github.com/sphinx-doc/sphinx/blob/master/sphinx/builders/latex/__init__.py#L439)) each time `app.config.latex_documents` is read. Thus, this assignment of `app.config.latex_documents[0]` is no-op.

**To Reproduce**

* Add `print(app.config.latex_documents)` before and after  `app.config.latex_documents[0] = app.config.latex_documents[0][:4] + (docclass,)` statement in `tests/test_build_latex.py::test_build_latex_doc`
* Launch the testsuite with :
```
tox -e py37 tests/test_build_latex.py::test_build_latex_doc -- -s
```
in order to see the debug prints.
Here are my [logs](https://github.com/sphinx-doc/sphinx/files/3511068/debug.log)

**Expected behavior**
We expect that the testsuite checks the different documentclasses properly.
But is `__getattr__` working as you expect ?
I am ready to help you on this but I need your insights.

**Environment info**
- OS: Linux
- Python version: 3.7.4
- Sphinx version: tested on master (commit 47cd262b3e) and 2.1.2 release
- Sphinx extensions:  None
- Extra tools: None

**Comment by tk0miya** [comment]
You're right! The statement does not modify anything.

**Comment by tk0miya** [comment]
Fixed by #6663 .
Thank you for reporting!
